packages/cli/src/dev.ts

```diff
@@ -0,0 +1,143 @@
+import Ajv, { Schema } from 'ajv';
+import addFormats from 'ajv-formats';
+import chokidar from 'chokidar';
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+import { Miniflare } from 'miniflare';
+import ora from 'ora';
+import * as path from 'path';
+
+import { buildScriptFromManifest } from './build';
+import { prettyPath } from './files';
+import { getDefaultManifestPath, resolveFile, resolveIntegrationManifestPath } from './manifest';
+import { getAPIClient } from './remote';
+import { createDevTunnel } from './tunnel';
+
+export const GITBOOK_DEV_SERVER_PORT = 8787;
+export const GITBOOK_DEV_CONFIG_FILE = 'gitbook-dev.yaml';
+
+const spinner = ora({ color: 'blue' });
+
+/**
+ * Start the integrations dev server on the given port. If no port is given,
+ * the default one is used. The dev server will automatically reload changes to
+ * the integration script.
+ */
+export async function startIntegrationsDevServer(port: number = GITBOOK_DEV_SERVER_PORT) {
+    spinner.start('Starting dev server...');
+    /**
+     * Read the integration manifest and build the integration script so that
+     * it can be served by the dev server.
+     * Also, read the dev config file to get the space to use for the dev server.
+     */
+    const manifestSpecPath = await resolveIntegrationManifestPath(getDefaultManifestPath());
+    const devConfigPath = resolveFile(manifestSpecPath, GITBOOK_DEV_CONFIG_FILE);
+    const devConfig = await readDevConfig(devConfigPath);
+    const { path: scriptPath, manifest } = await buildScriptFromManifest(manifestSpecPath);
+
+    /**
+     * Create a tunnel to allow the dev server to receive integration events
+     * from the GitBook platform
+     */
+    const tunnelUrl = await createDevTunnel(port);
+
+    /**
+     * Start the miniflare dev server. It will automatically reload the script
+     * when it detects a change with watch mode
+     */
+    const mf = new Miniflare({
+        scriptPath,
+        port,
+        liveReload: true,
+        watch: true,
```

Comment:
You are watching the script generated by esbuild, but not the source.

You need your own watcher here, or you need to have miniflare triggers the build (like https://miniflare.dev/developing/esbuild)

Reply:
Missed it below!